iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -188,6 +197,31 @@ private static ExtensionXmlSerializer getExtensionXmlSerializer(ExportOptions op
         return extensionXmlSerializer;
     }
 
+
+    private static XMLStreamWriter getConvenientWriter(NetworkXmlWriterContext context) {
```

Comment:
Instead of checking here if `context.getExtensionsWriter()` is null, initialize it as equal to `writer` when you create your context. That way, it will never be `null` when you call it.